### DISCUSSION TITLE
Video Player Seek Bar

### DISCUSSION THREAD

**Discussion by esegebart** [react-native-tvos/react-native-tvos#461] [Q&A] [answered]
I am curious if anyone has built a seekbar that works with react native tvos so the remote can skip forward on the slider. I know that there isn't a right and left arrow listener and only select, but if someone has accomplished this, what did you do?

I have very long video streams and right now I can set the time the user skips ahead - but if what they are watching is starting 8 hours into the stream, skipping ahead 30s or 5 minutes at a time isn't feasible. So I am curious how to get the focus on the slider thumb and skip forward maybe in progressive speeds like 2x, 3x, 4x. Thank you!

**Comment by coucoseth** [comment, marked as answer]
i am currently working on this same exact feature rightnow, i will update you incase i am successful, hopefully u'll do the same incase you have a solution before i do

**Reply by esegebart** [reply]
Yes absolutely! Good luck :) I have an idea - I am hoping to figure it out by week's end.

**Reply by altaywtf** [reply]
👋 hey there! I'm on the same ship. I've been playing with `@react-native-community/slider` but it doesn't work as good as I want with TV's remote. have you figured it out? 🤓

**Reply by coucoseth** [reply]
one year later after skipping this feature in our app, its here to haunt me, now i must do it. Anyone figured anything out yet?

**Reply by douglowder** [reply]
Probably you want to have left and right buttons that look for long press events and keep increasing the seek interval. I have a basic video example up using `expo-av` , maybe I can have a look to figure out how to modify that example to do something like this. https://github.com/douglowder/VideoTest/blob/main/App.tsx

**Reply by kosso** [reply]
I recently had to build something to do this and pretty did a very similar method to @douglowder there. 

I'm very new to React itself, so I might have made some mistakes, but it seems to all work nicely. (so far!)

I also included the ability to seek forward and back using the remote control/pad left/right (which will only do so when the progress bar 'thumb' has focus in the Controls.)

https://github.com/kosso/VideoPlayerDemo

(The full app I recently built actually just got in the Apple TV store too! :) Search for "Amazing TV" :+1: )
